resources/jenkins/release.sh

```diff
@@ -3,43 +3,37 @@ set -e
 
 # This script should be run after the build/tests have successfully completed
 
-MAJOR_MINOR=$1
-
 # Prune any local tags that aren't on the remote
 git fetch --prune origin "+refs/tags/*:refs/tags/*"
 
-DESCRIBE=`git describe --tags --always`
-REV=$(git log --pretty=format:'%h' -n 1)
+cd $WORKSPACE
 
-CURR_VERSION=`echo $DESCRIBE | awk '{split($0,a,"-"); print a[1]}'`
-echo "Got current git version $CURR_VERSION"
-MAJOR_VERSION=`echo $CURR_VERSION | awk '{split($0,a,"."); print a[1]}'`
-MINOR_VERSION=`echo $CURR_VERSION | awk '{split($0,a,"."); print a[2]}'`
+# Let's get version from maven
+MVNVER=`xmllint --xpath "/*[local-name()='project']/*[local-name()='version']/text()" pom.xml`
+TAG_NAME="v${MVNVER/-SNAPSHOT/}"
+echo $TAG_NAME
 
-if [ "$MAJOR_MINOR" == "Minor" ]; then
-    MINOR_VERSION=$((MINOR_VERSION+1))
-elif [ "$MAJOR_MINOR" == "Major" ]; then
-    MAJOR_VERSION=$((MAJOR_VERSION+1))
-    MINOR_VERSION=0
+if ! git rev-parse $TAG_NAME >/dev/null 2>&1
+then
+    git tag -a $TAG_NAME -m "Tagged automatically by Jenkins"
+	  git push origin $TAG_NAME
 else
-    echo "Error, unrecognized release type $MAJOR_MINOR.  Options are 'Major' and 'Minor'"
-    exit 1
+	echo "Tag: $TAG_NAME already exists."
 fi
 
-NEW_VERSION=$MAJOR_VERSION.$MINOR_VERSION
-echo "New version will be $NEW_VERSION"
+VERSION_FULL=`git describe --match "v[0-9\.]*" --long`
+echo "Full version: ${VERSION_FULL}"
 
-cd $WORKSPACE
+VERSION=`echo ${VERSION_FULL} | awk -F- '{print $1"-"$2}'`
+VERSION=${VERSION:1}
+echo "Package version: ${VERSION}"
 
-git tag -a $NEW_VERSION -m "New $MAJOR_MINOR release"
+REV=$(git log --pretty=format:'%h' -n 1)
+
+# bulding the debian package expects the file target/jibri.jar
+mv target/jibri-${MVNVER}-jar-with-dependencies.jar target/jibri.jar
 
-dch -v "$NEW_VERSION.$BUILD_NUMBER-1" "Built from git. $REV"
+dch -v "${VERSION}-1" "Built from git. $REV"
```

Comment:
shouldn't this have the short hash instead of the 1?

Reply:
"-1" is the debian version. Normally that is used for packages in the official repos, when you just change something in the debian subfolder and you bump it, and normally that is the only source that is kept on debian version control servers, away from upstream. Updating upstream version always starts from "-1" debian version.
The actual hash from git describe command is only used when versioning the maven artifacts. There is a command to do the other way around I think ... hum... need to check how we can convert from 8.0-3 to hash.